src/vmm_mad/remotes/lib/lxd/client.rb

```diff
@@ -130,7 +130,14 @@ def get_response(request, data)
 
         response.reading_body(@socket, request.response_body_permitted?) {}
 
-        response = JSON.parse(response.body)
+        begin
+            response = JSON.parse(response.body)
+        rescue StandardError => e
+            raise e unless e.message.include?('A JSON text must at least'\
+                'contain two octets! (JSON::ParserError)')
+
+            response.reading_body(@socket, request.response_body_permitted?) {}
```

Comment:
If it is other error we are reading again, but the resulting `response` is not parsed